test/models/user_test.rb

```diff
@@ -718,11 +718,11 @@ def setup_user(operation, type = 'users', search = nil)
     not_existing_auth_source = 'new_external_source'
 
     context "internal or not existing AuthSource" do
-      test 'existing user' do
+      test 'existing user without auth source specified' do
         assert_difference('User.count', 0) do
           login = users(:one).login
-          assert_equal User.find_or_create_external_user({:login => login}, nil),
-            User.find_by_login(login)
+          user = User.find_or_create_external_user({:login => login}, nil)
+          assert user.nil?
         end
       end
```

Comment:
This test should test that login still works if the `authorize_login_delegation_auth_source_user_autocreate` setting is disabled, the change that is needed is to use `login = users(:external).login` instead and verify the login is successful